File: tests/testdata/codegen/depends-on-array-pp/go/depends-on-array.go
package main

import (
	"fmt"

	"github.com/khulnasoft/khulnasoft-aws/sdk/v5/go/aws/s3"
	"github.com/khulnasoft/khulnasoft/sdk/v3/go/khulnasoft"
)

func main() {
	khulnasoft.Run(func(ctx *khulnasoft.Context) error {
		myBucket, err := s3.NewBucket(ctx, "myBucket", &s3.BucketArgs{
			Website: &s3.BucketWebsiteArgs{
				IndexDocument: khulnasoft.String("index.html"),
			},
		})
		if err != nil {
			return err
		}
		ownershipControls, err := s3.NewBucketOwnershipControls(ctx, "ownershipControls", &s3.BucketOwnershipControlsArgs{
			Bucket: myBucket.ID(),
			Rule: &s3.BucketOwnershipControlsRuleArgs{
				ObjectOwnership: khulnasoft.String("ObjectWriter"),
			},
		})
		if err != nil {
			return err
		}
		publicAccessBlock, err := s3.NewBucketPublicAccessBlock(ctx, "publicAccessBlock", &s3.BucketPublicAccessBlockArgs{
			Bucket:          myBucket.ID(),
			BlockPublicAcls: khulnasoft.Bool(false),
		})
		if err != nil {
			return err
		}
		_, err = s3.NewBucketObject(ctx, "index.html", &s3.BucketObjectArgs{
			Bucket:      myBucket.ID(),
			Source:      khulnasoft.NewFileAsset("./index.html"),
			ContentType: khulnasoft.String("text/html"),
			Acl:         khulnasoft.String("public-read"),
		}, khulnasoft.DependsOn([]khulnasoft.Resource{
			publicAccessBlock,
			ownershipControls,
		}))
		if err != nil {
			return err
		}
		ctx.Export("bucketName", myBucket.ID())
		ctx.Export("bucketEndpoint", myBucket.WebsiteEndpoint.ApplyT(func(websiteEndpoint string) (string, error) {
			return fmt.Sprintf("http://%v", websiteEndpoint), nil
		}).(khulnasoft.StringOutput))
		return nil
	})
}

File: tests/testdata/codegen/depends-on-array-pp/python/depends-on-array.py
import khulnasoft
import khulnasoft_aws as aws

my_bucket = aws.s3.Bucket("myBucket", website={
    "index_document": "index.html",
})
ownership_controls = aws.s3.BucketOwnershipControls("ownershipControls",
    bucket=my_bucket.id,
    rule={
        "object_ownership": "ObjectWriter",
    })
public_access_block = aws.s3.BucketPublicAccessBlock("publicAccessBlock",
    bucket=my_bucket.id,
    block_public_acls=False)
index_html = aws.s3.BucketObject("index.html",
    bucket=my_bucket.id,
    source=khulnasoft.FileAsset("./index.html"),
    content_type="text/html",
    acl="public-read",
    opts = khulnasoft.ResourceOptions(depends_on=[
            public_access_block,
            ownership_controls,
        ]))
khulnasoft.export("bucketName", my_bucket.id)
khulnasoft.export("bucketEndpoint", my_bucket.website_endpoint.apply(lambda website_endpoint: f"http://{website_endpoint}"))
